components/postmark/postmark.app.mjs

```diff
@@ -1,11 +1,208 @@
+      label: "Attachments",
+      description: `Each attachment should be a string with the parameters separated by a pipe character \`|\`, in the format: \`Name|Content|ContentType\`. Alternatively, you can pass a string representing an object. All three parameters are required:
+      \\
+      \\
+      \`Name\` - the filename with extension, i.e. \`readme.txt\`
+      \\
+      \`Content\` - the base64-encoded string with the binary data for the file, i.e. \`dGVzdCBjb250ZW50\`
+      \\
+      \`ContentType\` - the MIME content type, i.e. \`text/plain\`
+      \\
+      \\
+      Example with pipe-separated parameters: \`readme.txt|dGVzdCBjb250ZW50|text/plain\`
+      \\
+      Example with JSON-stringified object: \`{"Name":"readme.txt","Content":"dGVzdCBjb250ZW50","ContentType":"text/plain"}\`
+      `,
+      optional: true,
+    },
+    metadata: {
+      type: "object",
+      label: "Metadata",
+      description: "Custom metadata key/value pairs.",
+      optional: true,
+    },
+    message_stream: {
+      type: "string",
+      label: "Message stream",
+      description:
+        "Set message stream ID that's used for sending. If not provided, message will default to the outbound transactional stream.",
+      optional: true,
+    },
+  },
   methods: {
     // this.$auth contains connected account data
     authKeys() {
       console.log(Object.keys(this.$auth));
     },
+    async listTemplates() {
+      return await axios({
+        url: "https://api.postmarkapp.com/templates?count=500&offset=0",
+        headers: this.getHeaders(),
+        method: "GET",
+      })
+        .then(({ data }) =>
+          data.TotalCount
+            ? data.Templates.filter(
+              (obj) => obj.TemplateType === "Standard",
+            ).map((obj) => {
+              return {
+                label: obj.Name,
+                value: obj.Alias,
+              };
+            })
+            : [
+              {
+                label:
+                    "No templates found for this Postmark account. Create a template, then refresh this field.",
+                value: -1,
+              },
+            ])
+        .catch(() => [
+          {
+            label:
+              "An error ocurred while fetching the list of templates for this Postmark account. Please try again.",
+            value: -2,
+          },
+        ]);
+    },
+    getHeaders() {
+      return {
+        "X-Postmark-Server-Token": this.$auth.api_key,
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+      };
+    },
+    listSharedProps() {
+      return [
+        "from_email",
+        "to_email",
+        "cc_email",
+        "bcc_email",
+        "tag",
+        "reply_to",
+        "custom_headers",
+        "track_opens",
+        "track_links",
+        "attachments",
+        "metadata",
+        "message_stream",
+      ];
+    },
+    async sharedActionRequest($, action, endpoint, uniqueProps) {
```

Comment:
I think having a shared action request method is great, but also have specific methods for each action, for example:

```js
postmark.mjs

async sendEmailWithTemplate($, data) {
  return this.sharedActionRequest($, endpoint = "email/withTemplate", data);
}
```

And inside the action:

```js
send-email-with-template.mjs

async run({ $ }) {
  ...
  const response = await this.postmark.sendEmailWithTemplate($, data);
  $.export("$summary", "Some summary here");
  return response;
}
```

Reply:
Created individual methods for each of the two actions on the `app`.